File: api-dot-net/Dto/RequestDto/CreateProblemDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using CjsApi.Models;

namespace CjsApi.Dto
{
    public class CreateProblemDto
    {
        [Required]
        [MaxLength(150)]
        public string Title { get; set; } = null!;

        [Required]
        public string Description { get; set; } = null!;

        [Required]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Difficulty Difficulty { get; set; }

        [Range(100, 10000)]
        public int TimeLimitMs { get; set; } = 1000;

        [Range(64, 1024)]
        public int MemoryLimitMb { get; set; } = 1024;

        public bool IsPublished { get; set; } = true;

        // Tags sent as simple strings
        public List<string> Tags { get; set; } = new();

        // Test cases
        public List<TestCaseDto> TestCases { get; set; } = new();
    }
}

File: java-api/cjs-service/src/main/java/com/cjs/cjs_service/dto/request/CreateProblemDto.java
package com.cjs.cjs_service.dto.request;

import com.cjs.cjs_service.dto.TestCaseDto;
import com.cjs.cjs_service.model.Difficulty;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class CreateProblemDto {

    @NotBlank
    @Size(max = 150)
    private String title;

    @NotBlank
    private String description;

    @NotNull
    private Difficulty difficulty;

    @Min(100)
    @Max(10000)
    private int timeLimitMs = 1000;

    @Min(64)
    @Max(1024)
    private int memoryLimitMb = 1024;

    private boolean isPublished = true;

    // Tags as simple strings
    private List<String> tags = new ArrayList<>();

    // Test cases
    private List<TestCaseDto> testCases = new ArrayList<>();

    // getters and setters
}
